client/app/pages/queries/hooks/useAutoLimitFlags.js

```diff
@@ -0,0 +1,24 @@
+import { useCallback, useMemo, useState } from "react";
+import localOptions from "@/lib/localOptions";
+import { extend } from "lodash";
+
+function isAutoLimitAvailable(dataSource) {
+  return dataSource !== null && dataSource.apply_auto_limit;
+}
+
+export default function useAutoLimitFlags(dataSource, query, setQuery) {
+  const isAvailable = isAutoLimitAvailable(dataSource);
+  const [isChecked, setIsChecked] = useState(localOptions.get("ApplyAutoLimit", true));
+  query.options.applyAutoLimit = isAvailable && isChecked;
+
+  const checkboxAutoLimit = useCallback(
```

Comment:
Maybe call it `setAutoLimit` or `toggleAutoLimit`? It looks more like the action that will be done.